geopandas/plotting.py

```diff
@@ -171,14 +184,11 @@ def plot_linestring_collection(
 
     # color=None gives black instead of default color cycle
     if color is not None:
-        if is_color_like(color):
-            kwargs["color"] = color
-        elif pd.api.types.is_list_like(color):
-            kwargs["color"] = np.take(color, multiindex, axis=0)
-        else:
-            raise TypeError(
-                "Color attribute has to be a single color or sequence of colors."
-            )
+        kwargs["color"] = color
+        _check_and_expand(kwargs, "color", multiindex, is_color_like)
+
+    if "linewidth" in kwargs:
```

Comment:
Can we do it more generally? There is a large number of options we can pass to matplotlib, I would prefer once solution covering them all instead of specifying them one by one. You mentioned that `alpha` does not work as list-like, so there should be some check on the way (maybe even try/except?).

Reply:
I agree the goal should be to check/expand them all. There is a few checks I can imagine, but the easiest would be, to apply only the "expansion" to each keyword in `kwargs`, and then pass them on to `mpl`. So: if it is list-like, expand it; if not, just pass it on as-is. 

The advantages: we don't need to know anything a priori about these keywords (like, what their allowed values are etc.), our code is cleaner/more concise, and we don't replicate anything that `mpl` already does. We can just let any exceptions bubble up. 
This would also apply to / work for the `alpha` case I mentioned.

The only keyword where a check *is* necessary is, where there is some overlap/logic that needs to be done, such as in the case of `color` vs `facecolor`/`edgecolor`. 

The more I think about it, the more sense it makes. What do you think?

Reply:
That sounds sensible, but might get tricky.

`color`/`facecolor` logic is handled separately now anyway, we can keep it separately. 

> This would only work if the parameter in question does not accept a list-like "single" value

This is the tricky part. There are style kwds for which we would need to use specific checks, because if it supports list-like single value, we need to repeat that, like

- `color` (and its variants), but that should be already covered
- [linestyle](https://matplotlib.org/3.1.0/gallery/lines_bars_and_markers/linestyles.html) accepts tuple
- `markevery` accepts tuple/list

There's likely more. But I think it is worth exploring.

Reply:
I had another idea. As a function like `is_linestyle_like` does not seem to exist, how about the following.
For each key-value-pair in `kwargs` with which e.g. `plot_linestring_collection` is called, an line is tried to be created like so:

```python
from matplotlib.lines import Line2D
if 'linestyle' in kwargs:
    try:
        l = Line2D([0],[0],linestyle=kwargs['linestyle'])
    except ValueError:
        # expand
```

My only doubt is with performance.

Do you see any objections? If not I'll go and implement it like this.

Reply:
That does not look like an optimal solution. It comes with an overhead and fallback strategy is never a good way to go.

If the issue is `linestyle` and `markevery` only, we can find a way how to check them. `linestyle` accepts tuple of specific shape `(offset, (on_off_seq))` which can be checked for easily and `markevery` is probably not even applicable to our plots.

So I would add check for `is_linestyle_like` and go forward with two special cases (this and color).

This should do.
```py
def is_linestyle_like(ls):
    if len(ls) == 2 and isinstance(ls, tuple) and isinstance(ls[1], tuple):
        return True
    return False
```

Reply:
Yes, performance was also my concern.

Ok I've implemented it as suggested, with checks on the specific form that these single list-like values can take. 

Also, if an attribute is not allowed to be list-like (`alpha` for example, and `marker` - a list of alphas, one for each element in the `GeoDataFrame`, is not accepted the way a list of linewidths is - ([see docs](https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.axes.Axes.scatter.html#matplotlib.axes.Axes.scatter))), this attribute is no longer expanded, even *if* it's list-like - an error will be thrown either way. 

I've gone through the docs as well as the source code, and verified the most common attributes, and it's an improvement over how it was, but I can't guarantee I've been 100% exhaustive.